module/zfs/zap_micro.c

```diff
@@ -1072,6 +1072,21 @@ zap_prefetch(objset_t *os, uint64_t zapobj, const char *name)
 	return (err);
 }
 
+int
+zap_prefetch_object(objset_t *os, uint64_t zapobj)
+{
+	int error;
+	dmu_object_info_t doi;
+
+	error = dmu_object_info(os, zapobj, &doi);
+	if (error == 0 && DMU_OT_BYTESWAP(doi.doi_type) != DMU_BSWAP_ZAP)
```

Comment:
Do we really care about byteswap or anything else here except size?

Reply:
Looks like other places in that file are confirming raw type is always DMU_BSWAP_ZAP.  Not sure how we would end up here with it being some other type.

Reply:
I am not saying we may get here with some other type. But other places in that file may have logic specific to ZAP, while the prefetch is pretty universal.  But whatever.

Reply:
The zap changes were committed via PR #15888